components/camel-mail/src/main/java/org/apache/camel/component/mail/MailConsumer.java

```diff
@@ -388,63 +387,57 @@ private SearchTerm computeSearchTerm() {
+            if (!message.getFlags().contains(Flags.Flag.DELETED)) {
+                Exchange exchange = createExchange(message);
+                if (getEndpoint().getConfiguration().isMapMailMessage()) {
+                    // ensure the mail message is mapped, which can be ensured by touching the body/header/attachment
+                    LOG.trace("Mapping from javax.mail.Message to Camel MailMessage");
+                    exchange.getIn().getBody();
+                    exchange.getIn().getHeaders();
+                    // must also map attachments
+                    try {
+                        Map<String, Attachment> att = new HashMap<>();
+                        getEndpoint().getBinding().extractAttachmentsFromMail(message, att);
+                        if (!att.isEmpty()) {
+                            exchange.getIn(AttachmentMessage.class).setAttachmentObjects(att);
                         }
+                    } catch (MessagingException | IOException e) {
+                        // must release exchange before throwing exception
+                        releaseExchange(exchange, true);
+                        throw new RuntimeCamelException("Error accessing attachments due to: " + e.getMessage(), e);
                     }
+                }
 
-                    // If the protocol is POP3 we need to remember the uid on the exchange
-                    // so we can find the mail message again later to be able to delete it
-                    // we also need to remember the UUID for idempotent repository
-                    exchange.setProperty(MAIL_MESSAGE_UID, key);
+                // If the protocol is POP3 we need to remember the uid on the exchange
+                // so we can find the mail message again later to be able to delete it
+                // we also need to remember the UUID for idempotent repository
+                exchange.setProperty(MAIL_MESSAGE_UID, key);
 
-                    answer.add(exchange);
-                } else {
-                    if (LOG.isDebugEnabled()) {
-                        LOG.debug("Skipping message as it was flagged as deleted: {}", MailUtils.dumpMessage(message));
-                    }
-                }
-            } catch (Exception e) {
-                if (skipFailedMessage) {
-                    LOG.debug("Skipping failed message at index {} due {}", i, e.getMessage(), e);
-                } else if (handleFailedMessage) {
-                    handleException(e);
-                } else {
-                    throw e;
+                return exchange;
+            } else {
+                if (LOG.isDebugEnabled()) {
+                    LOG.debug("Skipping message as it was flagged as deleted: {}", MailUtils.dumpMessage(message));
                 }
             }
+        } catch (Exception e) {
+            if (skipFailedMessage) {
+                LOG.debug("Skipping failed message at index due {}", e.getMessage(), e);
```

Comment:
There is no longer an index so this log message should be updated

Reply:
Done